### ISSUE TITLE
Command line - compose middleware in deps.edn

### ISSUE THREAD

**Issue by arichiardi** [nrepl/nREPL#55]
Hello folks!

I think that, not nRepl's fault for sure, one thing that does not compose well is the `--middleware` option at the moment. Say I have a `:cider-clj` entry in `deps.edn` with:

```clojure
:main-opts ["-m" "nrepl.cmdline" "--middleware" "[cider.piggieback/wrap-cljs-repl]"]
```

In theory I would like to add just one thing, the piggieback middleware, without changing anything else. Instead we have to add a `:cider-cljs` entry with:

```clojure
:main-opts ["-m" "nrepl.cmdline" "--middleware" "[cider.nrepl/cider-middleware,cider.piggieback/wrap-cljs-repl]"]
```

Maybe we could allow many `--middleware` options and drop the vector format. If `deps.edn` conjoins vectors (will verify), probably we can just append things at the tail or `:main-opts`.

**Comment by bbatsov** [comment]
That's fine by me. I didn't fret too much over this when I added it - it was simply the behaviour we had for a while in `cider-nrepl` and it seemed to work ok. One small frustration I have with the cli options is that you can have `-m` for main and then `-m` for middleware, but I guess most people will sort this out. :-) 

 PRs welcome!

**Comment by bbatsov** [comment]
That's also somewhat related to https://github.com/nrepl/nREPL/issues/54 I guess.

**Comment by arichiardi** [comment]
Yep, they are totally related and yes, there is still a lot of work when it comes to composing options in `deps.edn`. For instance one thing you need to be careful when you have these kinds of `:main-opts` aliases is that with `-A` you will include more than one, in case two aliases both have `:main-opts`.

I am almost tempted actually to delegate this kind of aliasing to something more suitable like `.bash_aliases` for instance. It seems to me that command aliases is still rough in `deps.edn`.

**Comment by bbatsov** [comment]
Btw, now that #66 is merged you can also specify some default middleware via `.nrepl/config.edn` or `.nrepl-config.edn`. I think that's a much better approach than having each build tool have their own way to configure nREPL, although this will also require small updates to `boot` and `lein`'s REPL tasks.

**Comment by blak3mill3r** [comment]
For posterity, the global config file name is `.nrepl/nrepl.edn` (not `.nrepl/config.edn`). That confused me for a moment.

**Comment by bbatsov** [comment]
@blak3mill3r Yeah. The conversation here happened before we had settled on the final name.
